honeybee_radiance/state.py

```diff
@@ -0,0 +1,744 @@
+            return self.modifier
+        return self._modifier_direct
+
+    @modifier_direct.setter
+    def modifier_direct(self, value):
+        if value is not None:
+            assert isinstance(value, Modifier), \
+                'Expected Modifier for RadianceState. Got {}'.format(type(value))
+            value.lock()  # lock editing in case modifier has multiple references
+        self._modifier_direct = value
+
+    @property
+    def vmtx_geometry(self):
+        """Get or set a Face3D to be used for the inward-facing vmtx file.
+
+        If None, it will be a flipped (inward-facing) version of this state's parent.
+        Note that this property is only used in 3-phase and 5-phase studies and
+        its usual purpose is to account for thickness of the tmtx (BSDF) layer.
+        Also note that the vmtx_offset or tmtx_thickness methods can be used to
+        automatically generate this geometry without the need to set it here.
+        """
+        if not self._vmtx_geometry:  # use the inward-version of the parent geometry
+            return self.parent.geometry.flip() if self.has_parent else None
+        return self._vmtx_geometry
+
+    @vmtx_geometry.setter
+    def vmtx_geometry(self, value):
+        if value is not None:
+            assert isinstance(value, Face3D), \
+                'Expected Face3D for RadianceSubFaceState vmtx_geometry. ' \
+                    'Got {}'.format(type(value))
+        self._vmtx_geometry = value
+
+    @property
+    def dmtx_geometry(self):
+        """Get or set a Face3D to be used for the outward-facing dmtx file.
+
+        If None, it will be a flipped (inward-facing) version of this state's parent.
+        Note that this property is only used in 3-phase and 5-phase studies and
+        its usual purpose is to account for thickness of the tmtx (BSDF) layer.
+        Also note that the dmtx_offset or tmtx_thickness methods can be used to
+        automatically generate this geometry without the need to set it here.
+        """
+        if not self._dmtx_geometry:  # use the inward-version of the parent geometry
+            return self.parent.geometry.flip() if self.has_parent else None
+        return self._dmtx_geometry
+
+    @dmtx_geometry.setter
+    def dmtx_geometry(self, value):
+        if value is not None:
+            assert isinstance(value, Face3D), \
+                'Expected Face3D for RadianceSubFaceState dmtx_geometry. ' \
+                    'Got {}'.format(type(value))
+        self._dmtx_geometry = value
+
+    @property
+    def mtxs_default(self):
+        """Get a boolean noting whether the vmtx_geometry and dmtx_geometry are None.
+        
+        This indicates that the vmtx_geometry and dmtx_geometry are both just a
+        flipped version of the parent geometry.
+        """
+        return self._vmtx_geometry is None and self._dmtx_geometry is None
+
+    def tmtx_thickness(self, thickness):
```

Comment:
I was going to call it `set_tmtx_thickness` but then I remembered that, if we are starting methods with `get_` or `set_`, then this is probably unnecessary. Do you want me to put the `set_` back?  Or do you think that `from_tmtx_thickness` makes sense when this is not a `classmethod`?  Or do you want to use something really explicit like  `generate_mtx_geometry_from_tmtx_thickness`?

Reply:
Maybe it will be easier if we do not use mtx in the name. We are setting up the thickness of the subface/aperture. I like the explicit approach more. Something like `gen_geos_from_thickness`?

Reply:
We definitely need some way to indicate which geometry is changed by the 3 different methods here. So, until we figure out better names, I think we have to keep `tmtx`, `vmtx`, and `dmtx`. But we can shorten it based on what you said. So the 3 methods will be:
`gen_geos_from_tmtx_thickness`
`gen_geo_from_vmtx_offset`
`gen_geo_from_dmtx_offset`

When you decide what name you want to change these `mtx` variables to, you can make the PR across all of the radiance folder repos and this repo. Personally, I think the current situation is fine but you and @rpg777 are the experts so it's ultimately up to you.